optimum/habana/transformers/generation/utils.py

```diff
@@ -957,7 +957,7 @@ def generate(
                             model_kwargs[other_inputs] = torch.nn.functional.pad(
                                 model_kwargs[other_inputs],
                                 (0, generation_config.max_new_tokens),
-                                value=generation_config.pad_token_id,
+                                value=0,
```

Comment:
Could you explain what is the generation_config.pad_token_id and why it needs reset to 0?

Reply:
["attention_mask", "token_type_ids" -> this should be set to 0.

Reply:
The `generation_config.pad_token_id` is 32001. And the original value for padding here is `0` before this PR: [L963](https://github.com/huggingface/optimum-habana/commit/d28bae9277b1078512a60a9f3c793aede2c873fc#diff-284d6c109e6788a4405e302113203f6b7e98be68afc7dd3d85b89c6329b9275eL963).